demos/common/python/models/hpe_hrnet.py

```diff
@@ -0,0 +1,136 @@
+        except ValueError:
+            self.nms_heatmaps_blob_name = self.heatmaps_blob_name
+        self.embeddings_blob_name = find_layer_by_name('embeddings', self.net.outputs)
+
+        self.output_scale = self.net.input_info[self.image_blob_name].input_data.shape[-1] / self.net.outputs[self.heatmaps_blob_name].shape[-1]
+
+        if target_size is None:
+            h, w = self.net.input_info[self.image_blob_name].input_data.shape[-2:]
+            target_size = min(h, w)
+        if aspect_ratio >= 1.0:  # img width >= height
+            input_height, input_width = target_size, round(target_size * aspect_ratio)
+        else:
+            input_height, input_width = round(target_size / aspect_ratio), target_size
+        self.h = (input_height + size_divisor - 1) // size_divisor * size_divisor
+        self.w = (input_width + size_divisor - 1) // size_divisor * size_divisor
+        default_input_shape = self.net.input_info[self.image_blob_name].input_data.shape
+        input_shape = {self.image_blob_name: (default_input_shape[:-2] + [self.h, self.w])}
+        self.logger.info('Reshape net to {}'.format(input_shape))
+        self.net.reshape(input_shape)
+
+        self.decoder = AssociativeEmbeddingDecoder(
+            num_joints=self.net.outputs[self.heatmaps_blob_name].shape[1],
+            adjust=True,
+            refine=True,
+            delta=0.5,
+            max_num_people=30,
+            detection_threshold=0.1,
+            tag_threshold=1,
+            pose_threshold=prob_threshold,
+            use_detection_val=True,
+            ignore_too_much=False,
+            dist_reweight=True)
+        self.size_divisor = size_divisor
+
+    @staticmethod
+    def _get_inputs(net):
+        image_blob_name = None
+        for blob_name, blob in net.input_info.items():
+            if len(blob.input_data.shape) == 4:
+                image_blob_name = blob_name
+            else:
+                raise RuntimeError('Unsupported {}D input layer "{}". Only 4D input layers are supported'
+                                   .format(len(blob.shape), blob_name))
+        if image_blob_name is None:
+            raise RuntimeError('Failed to identify the input for the image.')
+        return image_blob_name
+
+    def preprocess(self, inputs):
+        img = resize_image(inputs, (self.w, self.h), keep_aspect_ratio=True)
+        h, w = img.shape[:2]
+        if not (self.h - self.size_divisor < h <= self.h and self.w - self.size_divisor < w <= self.w):
+            self.logger.warn("Chosen model aspect ratio doesn't match image aspect ratio")
+        resize_img_scale = np.array((inputs.shape[1] / w, inputs.shape[0] / h), np.float32)
+
+        pad = [(self.h - h + 1) // 2, (self.w - w + 1) // 2, (self.h - h) // 2, (self.w - w) // 2]
+        img = np.pad(img, ((pad[0], pad[2]), (pad[1], pad[3]), (0, 0)),
+                     mode='constant', constant_values=0)
```

Comment:
Is the way an image is padded (and reverse operation in `postprocess()`) the only difference between `HpeHRNet` and `HpeAssociativeEmbedding`? If so it may be better to add something like padding mode arg instead of the whole model class.

Reply:
Add padding mode and using one model class for `ae` and `hrnet` now. Please, check the new version.